hi-ml/src/health_ml/runner.py

```diff
@@ -0,0 +1,319 @@
+
+        return parser_result
+
+    def run(self) -> Tuple[LightningContainer, AzureRunInfo]:
+        """
+        The main entry point for training and testing models from the commandline. This chooses a model to train
+        via a commandline argument, runs training or testing, and writes all required info to disk and logs.
+
+        :return: a tuple of the LightningContainer object and an AzureRunInfo containing all information about
+            the present run (whether running in AzureML or not)
+        """
+        # Usually, when we set logging to DEBUG, we want diagnostics about the model
+        # build itself, but not the tons of debug information that AzureML submissions create.
+        logging_to_stdout(logging.INFO if is_local_rank_zero() else "ERROR")
+        initialize_rpdb()
+        self.parse_and_load_model()
+        azure_run_info = self.submit_to_azureml_if_needed()
+        self.run_in_situ(azure_run_info)
+        return self.lightning_container, azure_run_info
+
+    def submit_to_azureml_if_needed(self) -> AzureRunInfo:
+        """
+        Submit a job to AzureML, returning the resulting Run object, or exiting if we were asked to wait for
+        completion and the Run did not succeed.
+
+        :return: an AzureRunInfo object containing all of the details of the present run. If AzureML is not
+            specified, the attribute 'run' will None, but the object still contains helpful information
+            about datasets etc
+        """
+        root_folder = self.project_root
+        entry_script = Path(sys.argv[0]).resolve()
+        script_params = sys.argv[1:]
+
+        additional_conda_env_files = self.lightning_container.additional_env_files
+        if additional_conda_env_files is not None:
+            additional_env_files = [Path(f) for f in additional_conda_env_files]
+        else:
+            additional_env_files = None
+
+        conda_dependencies_files = get_all_environment_files(self.project_root,
+                                                             additional_files=additional_env_files)
+        pip_requirements_files = get_all_pip_requirements_files()
+
+        # Merge the project-specific depclass Experendencies with the packages and write unified definition
+        # to temp file. In case of version conflicts, the package version in the outer project is given priority.
+        temp_conda: Optional[Path] = None
+        if len(conda_dependencies_files) > 1 or len(pip_requirements_files) > 0:
+            temp_conda = root_folder / f"temp_environment-{uuid.uuid4().hex[:8]}.yml"
+            merge_conda_files(conda_dependencies_files, temp_conda, pip_files=pip_requirements_files)
+
+        # TODO: Update environment variables
+        environment_variables: Dict[str, Any] = {}
+
+        # get default datastore from provided workspace
+        workspace = get_workspace()
+        default_datastore = workspace.get_default_datastore().name
```

Comment:
It might be better if we let users configure the datastore name.

Reply:
We could add that as a field to the lightning container - after all, a folder name only makes sense in the context of a datastore